From c2b518929da809425ccf12f8746a485afae37e7a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Mon, 13 Apr 2020 15:41:54 +0200
Subject: [PATCH 1/2] Revert "app-cache: Fix cache for folder translations"

Existing folders as created by gnome-software (including the
default ones) all have the .directory suffix, so dropping
the suffix from the keys broke those folders.

This reverts commit 343b3351f172e112e1a70b54956ab3444140fbf4.

https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2623
---
 src/shell-app-cache.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/src/shell-app-cache.c b/src/shell-app-cache.c
index 9b33c7ac2..15d4734d0 100644
--- a/src/shell-app-cache.c
+++ b/src/shell-app-cache.c
@@ -109,17 +109,11 @@ load_folder (GHashTable *folders,
 
   while ((name = g_dir_read_name (dir)))
     {
-      g_autofree gchar *stripped_name = NULL;
       g_autofree gchar *filename = NULL;
       g_autoptr(GKeyFile) keyfile = NULL;
 
-      if (!g_str_has_suffix (name, ".directory"))
-        continue;
-
-      stripped_name = g_strndup (name, strlen (name) - strlen (".directory"));
-
       /* First added wins */
-      if (g_hash_table_contains (folders, stripped_name))
+      if (g_hash_table_contains (folders, name))
         continue;
 
       filename = g_build_filename (path, name, NULL);
@@ -134,8 +128,7 @@ load_folder (GHashTable *folders,
                                                      NULL, NULL);
 
           if (translated != NULL)
-            g_hash_table_insert (folders, g_steal_pointer (&stripped_name),
-                                 translated);
+            g_hash_table_insert (folders, g_strdup (name), translated);
         }
     }
 }
-- 
2.26.0


From 3fac0632a8fdb78a7082dd403cbf27dd62ed91b7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sun, 12 Apr 2020 17:15:48 +0200
Subject: [PATCH 2/2] appDisplay: Look up directory- instead of category
 translations

Translations are provided by .directory files, so trying to look
up a category name without the suffix will always fail.

Commit 343b3351f1 tried to fix this previously by changing the
saved keys, but that broke existing translatable folders.

Appending the .directory suffix for the lookup instead fixes the
issue without regressing non-custom folders.

https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2623
---
 js/ui/appDisplay.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index 99c052b79..08febc491 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -115,7 +115,8 @@ function _findBestFolderName(apps) {
     }, commonCategories);
 
     for (let category of commonCategories) {
-        let translated = Shell.util_get_translated_folder_name(category);
+        const directory = '%s.directory'.format(category);
+        const translated = Shell.util_get_translated_folder_name(directory);
         if (translated !== null)
             return translated;
     }
-- 
2.26.0

